appmonitoring/ts/src/server.ts

```diff
@@ -1,13 +1,24 @@
-﻿import * as fs from "fs";
-import * as https from "https";
+﻿import * as https from "https";
 import { ContentProcessor } from "./ContentProcessor.js";
 import { logger, Metrics } from "./LoggerWrapper.js";
 import { AppMonitoringConfigCR, IRootObject } from "./RequestDefinition.js";
 import { K8sWatcher } from "./K8sWatcher.js";
 import { AppMonitoringConfigCRsCollection } from "./AppMonitoringConfigCRsCollection.js"
+import fs from "fs";
+import { CertificateManager } from "./CertificateGenerator.js";
 
+const containerMode = process.env.CONTAINER_MODE;
+
+if ("secrets-manager".localeCompare(containerMode) === 0) {
+
+    logger.info("Running in certificate manager mode...");
+    await CertificateManager.CreateWebhookAndCertificates();
+
+    process.exit();
```

Comment:
Let's have a log indicating successful completion. Also, let's surround it with try/catch/finally so that we can have meaningful logging

Reply:
we do have that in the actual method at the end

Reply:
There's a bunch of stuff not covered by those .catch()'s (I've left comments to convert them to try/catch btw). Let's have another try/catch here to be sure we are catching everything.